packages/webpack-cli/lib/utils/cli-flags.js

```diff
@@ -6,6 +8,38 @@ const ADVANCED_GROUP = 'advanced';
 const DISPLAY_GROUP = 'stats';
 const ZERO_CONFIG_GROUP = 'zero-config';
 
+// Extract all the flages being exported from core. A list of cli flags generated by core
+// can be found here https://github.com/webpack/webpack/blob/master/test/__snapshots__/Cli.test.js.snap
+let flagsFromCore =
+    typeof cli !== 'undefined'
+        ? Object.entries(cli.getArguments()).map(([flag, meta]) => {
+              if (meta.simpleType === 'string') {
+                  meta.type = String;
+                  meta.usage = `--${flag} <value>`;
+              } else if (meta.simpleType === 'number') {
+                  meta.type = Number;
+                  meta.usage = `--${flag} <value>`;
+              } else {
+                  meta.type = Boolean;
+                  meta.negative = true;
+                  meta.usage = `--${flag}`;
+              }
+              return {
+                  ...meta,
+                  name: flag,
+                  group: 'core',
+              };
+          })
+        : [];
+
+// duplicate flags
+const duplicateFlags = ['entry', 'mode', 'stats', 'watch', 'target'];
+// regExp Flags
+const regExpFlags = ['module-expr-context-reg-exp', 'module-unknown-context-reg-exp', 'module-wrapped-context-reg-exp'];
+// filter all duplicate, broken and regExp flags
+const flagsToFilter = [...duplicateFlags, ...regExpFlags];
```

Comment:
Why we ignore `regexp` flags? We can implement `regexp: true` for some options and use `new RegExp(options.something)` and set it for webpack

Reply:
You can also pass these are strings to webpack processArguments, it will convert them to regexp.

The webpack processArguments method allows strings for all arguments and converts them to the correct format.